src/WebcamInput.jl

```diff
@@ -0,0 +1,978 @@
+### A Pluto.jl notebook ###
+# v0.19.6
+
+using Markdown
+using InteractiveUtils
+
+# This Pluto notebook uses @bind for interactivity. When running this notebook outside of Pluto, the following 'mock version' of @bind gives bound variables a default value (instead of an error).
+macro bind(def, element)
+    quote
+        local iv = try Base.loaded_modules[Base.PkgId(Base.UUID("6e696c72-6542-2067-7265-42206c756150"), "AbstractPlutoDingetjes")].Bonds.initial_value catch; b -> missing; end
+        local el = $(esc(element))
+        global $(esc(def)) = Core.applicable(Base.get, el) ? Base.get(el) : iv(el)
+        el
+    end
+end
+
+# ╔═╡ 25fc026c-c593-11ec-05c8-93e16f9dc527
+using HypertextLiteral
+
+# ╔═╡ cfb76adb-96da-493c-859c-ad24aa18437e
+using UUIDs: UUID, uuid4
+
+# ╔═╡ b3b59805-7062-463f-b6c5-1679133a589f
+using ColorTypes
+
+# ╔═╡ ba13c770-5937-4a34-9c49-04cbc672f89e
+using ImageShow, ImageIO
+
+# ╔═╡ da787bc2-ef53-4bab-926c-7ab8bfbd50a9
+using MarkdownLiteral: @mdx
+
+# ╔═╡ 31dff3d3-b3ee-426d-aec8-ee811820d842
+import AbstractPlutoDingetjes
+
+# ╔═╡ 5104aabe-43f7-451e-b4c1-68c0b345669e
+"""
+Converts [ImageData](https://developer.mozilla.org/en-US/docs/Web/API/ImageData) vector to an RGBA Matrix.
+
+Accepts a `Dict{Any, Any}` similar to the ImageData type, i.e. with keys
+ - `width`: width of the image
+ - `height`: height of the image
+ - `data`: image data - 4 elements per pixel, ``4*width*height`` total length
+
+"""
+function ImageDataToRGBA(d::Dict)
+	width = d["width"]
+	height = d["height"]
+	r = reshape(d["data"], 4, width, height) ./ 255
+	return [
+		RGBA(
+			r[1, w, h],
+			r[2, w, h],
+			r[3, w, h],
+			r[4, w, h]
+		) for
+			h in 1:height,
+			w in 1:width
+		]
```

Comment:
We could do that to prevent creating an extra array:
```suggestion
	return reinterpret(reshape, RGBA{eltype(r)}, r)
```
The resulting type is not `Array{RGBA{Float64},2}` but only `AbstractArray{RGBA{Float64},2}` tough.

Reply:
Thanks for the suggestion Paul! I ended up using all the tricks to get a zero-allocations version, including `N0f8` from FixedPointNumbers.jl (just like the return type of `load(png_path)` from ImageIO.jl).

[`109ca3e` (#210)](https://github.com/JuliaPluto/PlutoUI.jl/pull/210/commits/109ca3e0b57bc7f899044c15d1ffa118ded5c93c)

Reply:
I added an option `WebcamInput(; avoid_allocs=true)` to enable this functionality, and I have it **off by default**, i.e. it returns a `Matrix{RGBA{N0f8}}` by default. 

I think it is almost impossible to understand what is going on if you do `typeof(img)`, and you see:
```
PermutedDimsArray{RGBA{N0f8}, 2, (2, 1), (2, 1), Base.ReshapedArray{RGBA{N0f8}, 2, Base.ReinterpretArray{RGBA{N0f8}, 1, UInt8, Vector{UInt8}, false}, Tuple{}}}
```

Reply:
> I think it is almost impossible to understand what is going on if you would do typeof(img), and you see:
> PermutedDimsArray{RGBA{N0f8}, 2, (2, 1), (2, 1), Base.ReshapedArray{RGBA{N0f8}, 2, Base.ReinterpretArray{RGBA{N0f8}, 1, UInt8, Vector{UInt8}, false}, Tuple{}}}

Definitely!